src/dawn/SIR/SIR.cpp

```diff
@@ -174,4 +174,137 @@ std::shared_ptr<sir::StencilCall> sir::StencilCall::clone() const {
   return call;
 }
 
+bool SIR::operator==(const SIR& rhs) const {
+  bool retval = true;
+
+  if((Stencils.size() == rhs.Stencils.size()) &&
+     (StencilFunctions.size() == rhs.StencilFunctions.size())) {
+
+    if(Stencils.size() > 0) {
+      retval &= std::equal(Stencils.begin(), Stencils.end(), rhs.Stencils.begin(),
+                           sharedpointer_comparison<sir::Stencil>);
+    }
+
+    if(StencilFunctions.size() > 0) {
+      retval &=
+          std::equal(StencilFunctions.begin(), StencilFunctions.end(), rhs.StencilFunctions.begin(),
+                     sharedpointer_comparison<sir::StencilFunction>);
+    }
+
+    if(GlobalVariableMap != nullptr && rhs.GlobalVariableMap != nullptr) {
+      if(GlobalVariableMap->size() == rhs.GlobalVariableMap->size()) {
+        if(GlobalVariableMap->size() > 0) {
+          retval &=
+              pointerMapComparison(*(GlobalVariableMap.get()), *(rhs.GlobalVariableMap.get()));
+        }
+      } else {
+        return false;
+      }
+    } else {
+      return GlobalVariableMap == rhs.GlobalVariableMap;
```

Comment:
Why do we need this? This will simply compare the pointers

Reply:
We have equality if both Maps are nullptrs.
First [194] we check if both are not null (in oder to ensure sizes do not segfault) but if either is nullptr, we need both to be null and therefore we do this

Reply:
Alright, makes sense.